src/loader/loader.c

```diff
@@ -392,6 +393,40 @@ stop_workers_on_db_drop(DropdbStmt *drop_db_statement)
 	}
 }
 
+static bool
+database_is_template(const Oid db_oid)
+{
+	Relation	pg_database;
+	ScanKeyData entry[1];
+	SysScanDesc scan;
+	HeapTuple	dbtuple;
+	bool	is_template = false;
+
+	pg_database = table_open(DatabaseRelationId, AccessShareLock);
+	ScanKeyInit(&entry[0],
+				Anum_pg_database_oid,
+				BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(db_oid));
+	scan = systable_beginscan(pg_database, DatabaseOidIndexId, true,
+							  NULL, 1, entry);
+
+	dbtuple = systable_getnext(scan);
+
+	/* We assume that there can be at most one matching tuple */
+	if (!HeapTupleIsValid(dbtuple))
+		ereport(ERROR,
+				(errcode(ERRCODE_UNDEFINED_DATABASE),
+				 errmsg("database with OID \"%u\" does not exist",
+						db_oid)));
+	
+	is_template = ((Form_pg_database) GETSTRUCT(dbtuple))->datistemplate;
+
+	systable_endscan(scan);
+	table_close(pg_database, AccessShareLock);
+
+	return is_template;
+}
+
```

Comment:
I think the right thing to do here is check by `Form_pg_database->datallowconn` because you can have a template database (`Form_pg_database->datistemplate=true`) allowing connections.

Reply:
We don't run bgw on template databases so there is no point checking for connect-ability.

Reply:
Try the following:

CREATE DATABASE foo WITH ALLOW_CONNECTIONS=false;
CREATE DATABASE bar WITH TEMPLATE=foo;